src/GraphQL/Utilities/Federation/FederatedSchemaPrinter.cs

```diff
@@ -52,6 +52,10 @@ public string PrintFederatedDirectivesFromAst(IProvideMetadata type)
 
         public override string PrintObject(IObjectGraphType type)
         {
+            // Do not return an empty query type: "Query { }" as it is not valid as part of the sdl.
+            if (type != null && String.Equals(type.Name, "Query", StringComparison.Ordinal) && !type.Fields.Any(x => !IsFederatedType(x.ResolvedType.GetNamedType().Name)))
+               return String.Empty;
```

Comment:
Needs to be indented 4 spaces (not 3 spaces. )

Reply:
```suggestion
                return String.Empty;
```

Reply:
You can commit the suggestion right here by the way.

Reply:
I downloaded the dotnet tool used, ran the tool, it modified the file, I checked in the change and pushed it to my forked branch.